contrib/pg_tde/src/catalog/tde_principal_key.c

```diff
@@ -945,11 +940,94 @@ pg_tde_is_provider_used(Oid databaseOid, Oid providerId)
 		bool		used = principal_key != NULL && providerId == principal_key->keyInfo.keyringId;
 
 		LWLockRelease(tde_lwlock_enc_keys());
-
 		return used;
 	}
 }
 
+/*
+ * Verifies that all keys that are currently in use matches the keys available
+ * at the provided key provider. This is meant to be used before modifying an
+ * existing provider to ensure the new settings will provide the same keys as
+ * those that are already in use.
+ */
+void
+pg_tde_verify_provider_keys_in_use(GenericKeyring *proposed_provider)
```

Comment:
The parameter was a bit misleading to me at the start, but then I realized that this is the existing provider, with modified settings, but the same `keyring_id`. Maybe it could have a better name / documentation so that this is clear. (`modified_provider` maybe?)